src/components/CallView/Grid/Grid.vue

```diff
@@ -979,16 +954,23 @@ export default {
 
 			// add the model
 			if (!this.tempPromotedModels.includes(model)) {
-				const videoIndex = this.tilesWithVideoMask.indexOf(id)
-				if (videoIndex !== -1) {
-					this.tilesWithVideoMask.splice(videoIndex, 1)
-				}
-				const audioIndex = this.tilesWithAudioMask.indexOf(id)
-				if (audioIndex !== -1) {
-					this.tilesWithAudioMask.splice(audioIndex, 1)
+				// remove model from the order history if it exists
+				const modelIndex = this.promotedHistoryMask.indexOf(id)
+				if (modelIndex !== -1) {
+					this.promotedHistoryMask.splice(modelIndex, 1)
 				}
 
 				this.tempPromotedModels.unshift(model)
+				// add model to the beginning of the orderedVideos in its category
+				if (this.isModelWithVideo(model)) {
+					if (!this.promotedHistoryMask.includes(id)) {
+						this.promotedHistoryMask.unshift(id)
+					}
+				} else {
+					if (!this.promotedHistoryMask.includes(id)) {
+						this.promotedHistoryMask.unshift(id)
+					}
+				}
```

Comment:
You wanted to write this, right? =P

from Line 958-960 we already know it's not in the array, or remove it

```suggestion
				this.promotedHistoryMask.unshift(id)
```